tests/linearGaussian_npse_test.py

```diff
@@ -179,27 +182,94 @@ def test_npse_iid_inference(num_trials):
     prior_mean = zeros(num_dim)
     prior_cov = eye(num_dim)
     prior = MultivariateNormal(loc=prior_mean, covariance_matrix=prior_cov)
-    gt_posterior = true_posterior_linear_gaussian_mvn_prior(
-        x_o, likelihood_shift, likelihood_cov, prior_mean, prior_cov
-    )
-    target_samples = gt_posterior.sample((num_samples,))
 
-    inference = NPSE(prior, show_progress_bars=True)
+    # This check that our method to handle "general" priors works.
+    # i.e. if NPSE does not get a proper passed by the user.
+    prior_npse = prior if not with_improper_prior else None
+    inference = NPSE(prior_npse, show_progress_bars=True, sde_type=sde_type)
 
     theta = prior.sample((num_simulations,))
     x = linear_gaussian(theta, likelihood_shift, likelihood_cov)
 
     score_estimator = inference.append_simulations(theta, x).train(
-        training_batch_size=100,
+        training_batch_size=200, max_num_epochs=400
```

Comment:
do we still need this increased `num_epochs`? was not this fixed with the recent update on the convergence?

Reply:
Yeah, will have a look into this. The tests did, however, fail on the new setting (although not by much, I can just change the tolerance of the checks a bit). In general, these will be slightly more sensitive to a small change in approximation performance on 1 trial; it will propagate.